boot/zephyr/main.c

```diff
@@ -103,8 +107,7 @@ void main(void)
 
     os_heap_init();
 
-    boot_flash_device = device_get_binding(FLASH_DEV_NAME);
-    if (!boot_flash_device) {
+    if (!flash_device_get_binding(FLASH_DEV_NAME)) {
```

Comment:
IMO we should make public the zephyr flash_map API `get_flash_dev_from_id`  and use it here, separately for check each image sector driver.

Reply:
^^ I mena to change this in the zephyr-project in near future

Reply:
obviously keep this as is in this PR, I'm just thinking about preparing to use external flash in the feature, just want your opinion.